libs/ml/include/ml/core/graph.hpp

```diff
@@ -490,6 +492,36 @@ std::vector<TensorType> Graph<TensorType>::get_weights() const
   return std::move(ret);
 }
 
+template <typename TensorType>
+void Graph<TensorType>::set_weights(std::vector<TensorType> &new_weights)
+{
+  SizeType index = 0;
+  for (auto &t : trainable_nodes_)
```

Comment:
it's a bit pedantic but in these cases where you want to iterate over two objects I think it's preferable to use
auto it1 = obj1.begin();
auto it2 = obj2.begin();
while (it1.is_valid()) // or equivalent code for the relevant object type
{
    // do work
    ++ it1;
    ++it2;
}

rather than mixing the use of one iterator and one counter/incrementer.